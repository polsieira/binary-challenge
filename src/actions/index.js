export const checkIsLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
})

export const addExoplanets = (exoplanets) => ({
  type: 'ADD_EXOPLANETS',
  exoplanets
})

export const loginUser = (userInfo) => ({
  type: 'LOGIN_USER',
  userInfo
})

export const addFilters = (filters) => ({
  type: 'ADD_FILTERS',
  filters
})