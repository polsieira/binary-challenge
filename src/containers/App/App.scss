.App {
  background-color: rgb(235, 235, 235);
  text-align: center;
}