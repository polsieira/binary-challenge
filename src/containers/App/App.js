import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.scss';
import { Route } from 'react-router-dom';
import AccountManager from '../../components/AccountManager/AccountManager';
import Footer from '../../components/Footer/Footer';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { getData } from '../../apiCalls';
import { checkIsLoading, hasErrored } from '../../actions';

class App extends Component {

  async componentDidMount() {
    const { checkIsLoading, hasErrored } = this.props;
    //https://api.betterdoctor.com/2016-03-01/doctors?name=%3Aname&first_name=%3Afirst_name&last_name=%3Alast_name&query=%3Aquery&specialty_uid=%3Aspecialty_uid&insurance_uid=%3Aspecialty_uid&practice_uid=%3Apractice_uid&location=37.773%2C-122.413%2C100&user_location=37.773%2C-122.413&gender=male&
    var apiBaseUrl = "https://api.betterdoctor.com/2016-03-01/";
    const url = `${apiBaseUrl}doctors?location=37.773%2C-122.413%2C100&user_location=37.773%2C-122.413&skip=0&limit=10&user_key=1ac2abd6f94242154ff7cef2b1137a83`;
    try {
      checkIsLoading(true);
      const data = await getData(url);
      checkIsLoading(false);
      console.log(data);
      // save data to store with action
    } catch ({ message }) {
      checkIsLoading(false);
      hasErrored(message);
    }
  }

  render() {
    return (
      <main className='App'>
        <Route path='/account' render={() => (<AccountManager />)} />
        <Footer />
      </main>
    )
  }
}

export const mapStateToProps = ({ isLoading, error }) => ({
  isLoading,
  error
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    checkIsLoading,
    hasErrored
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
  hasErrored: PropTypes.func,
  checkIsLoading: PropTypes.func,
  error: PropTypes.string,
  isLoading: PropTypes.bool,
}







