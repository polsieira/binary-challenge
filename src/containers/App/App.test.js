import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { checkIsLoading, hasErrored, addExoplanets } from '../../actions';

jest.mock('../../apiCalls.js');

describe('App', () => {
  describe('App component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with isLoading and error', () => {
      let isLoading, error;
      const mockState = {
        isLoading,
        error
      };

      const expected = {
        isLoading,
        error
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;

    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('calls dispatch with setLoading', () => {
      const actionToDispatch = checkIsLoading(true);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.checkIsLoading(true);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with hasErrored', () => {
      const error = {
        message: 'This is an error'
      };
      const actionToDispatch = hasErrored(error);

      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.hasErrored(error);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('calls dispatch with a setMovies action', () => {
      const exoplanets = [{
        pl_name: 'Pluto',
        pl_hostname: 'Sun',
        pl_orbper: 2000,
        gaia_dist: 100,
        pl_disc: 2005
      },
      {
        pl_name: 'Mearth',
        pl_hostname: 'Msun',
        pl_orbper: 563,
        gaia_dist: 8,
        pl_disc: 2000
      }]
      const actionToDispatch = addExoplanets(exoplanets);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addExoplanets(exoplanets);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

  });

});

