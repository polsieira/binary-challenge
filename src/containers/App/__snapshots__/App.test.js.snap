// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should match the snapshot 1`] = `
<main
  className="App"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/exoplanets/:planet"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/(login|create-account)"
    render={[Function]}
  />
</main>
`;
